core/src/main/java/org/pentaho/di/connections/vfs/VFSConnectionManagerWrapper.java

```diff
@@ -0,0 +1,114 @@
+package org.pentaho.di.connections.vfs;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
```

Comment:
About the use of `@NonNull` and `@Nullable` annotations, @abryant-hv raised the following:

https://github.com/pentaho/pentaho-kettle/pull/9334#pullrequestreview-2071681357
> I'm basically ignoring the @NonNull/etc annotations. I am ambivalent about whether they are introduced, but I'm kinda surprised they were introduced without more discussion. I would expect that kind of thing would be more of a pentaho-wide discussion.

I think these help immensely when writing code, letting the editor warn you of common mistakes. These also allow expressing the developer's intentions more clearly, complementing documentation (when it exists). 

There is precedent in using these in other Pentaho projects.

First, there was use of `com.google.code.findbugs:jsr305`.
- References:  https://github.com/search?q=org%3Apentaho+jsr305+language%3A%22Maven+POM%22&type=code 
- Introduced before 2016 by: https://github.com/pentaho/pentaho-platform/commit/f62ba3bd44a4f89ee1e0557b190ac9eda475ac18#diff-d5690600fce34298dfc538cbeeac72e4f64cf20b4c5ccf0cad1fd17664af82b6R2010

The findbugs project was abandoned, though.

When considering using this on some new code, I discovered that findbugs had been abandoned, and found out about a drop-in/API compatible replacement, `com.github.spotbugs:spotbugs-annotations`, which, I started using instead.
- References: https://github.com/search?q=org%3Apentaho%20spotbugs-annotations&type=code
- Introduced around 2021: https://github.com/pentaho/pentaho-platform/pull/4931/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R227

Honestly, I don't remember who I checked this with at the time. I imagine that I felt legitimized for adding the compatible-API-successor of a library which was already being used.

I realize this is one of those hot issues: https://stackoverflow.com/questions/4963300/which-notnull-java-annotation-should-i-use.

Reply:
Follow-up conversation in slack: https://hitachivantara-eng.slack.com/archives/C0819BSJK/p1717407061357839.